tests/index.test.ts

```diff
@@ -187,169 +233,157 @@ describe('ParaSwap SDK', () => {
+      side: SwapSide.BUY,
+      options: { includeDEXS: 'Uniswap,UniswapV2,Balancer,Oasis' },
     });
-    test('Build_and_Send_Tx_BUY', async () => {
-      const destAmount = srcAmount;
-      const ratesOrError = await paraSwap.getRate(
-        srcToken,
-        destToken,
-        destAmount,
-        senderAddress,
-        SwapSide.BUY,
-        { includeDEXS: 'Uniswap,UniswapV2,Balancer,Oasis' }
-      );
-      const priceRoute = ratesOrError as OptimalRate;
-      const _srcAmount = new BigNumber(priceRoute.srcAmount)
-        .times(1.1)
-        .toFixed(0);
+    const _srcAmount = new BigNumber(priceRoute.srcAmount)
+      .times(1.1)
+      .toFixed(0);
 
-      const txOrError = await paraSwap.buildTx(
+    const txParams = await paraSwap.buildTx(
+      {
         srcToken,
         destToken,
-        _srcAmount,
+        srcAmount: _srcAmount,
         destAmount,
         priceRoute,
-        signer.address,
-        referrer,
-        undefined,
-        undefined,
-        undefined,
-        { ignoreChecks: true }
+        userAddress: signer.address,
+        partner: referrer,
+      },
+      { ignoreChecks: true }
+    );
+
+    const transaction = {
+      ...txParams,
+      gasPrice: '0x' + new BigNumber(txParams.gasPrice).toString(16),
+      gasLimit: '0x' + new BigNumber(5000000).toString(16),
+      value: '0x' + new BigNumber(txParams.value).toString(16),
+    };
+    const toContract = new ethers.Contract(destToken, erc20abi, ethersProvider);
+    const beforeFromBalance = await ethersProvider.getBalance(signer.address);
+    const beforeToBalance = await toContract.balanceOf(signer.address);
+
+    const txr = await signer.sendTransaction(transaction);
+    await txr.wait(1);
+    const afterFromBalance = await ethersProvider.getBalance(signer.address);
+    const afterToBalance = await toContract.balanceOf(signer.address);
+    expect(beforeFromBalance.gt(afterFromBalance)).toBeTruthy();
+    expect(beforeToBalance.lt(afterToBalance)).toBeTruthy();
+  });
+});
+
+describe.each([
+  ['fetchFetcher & ethersContractCaller', fetchFetcher, ethersContractCaller],
+  ['axiosFetcher & web3ContractCaller', axiosFetcher, web3ContractCaller],
+])(
+  'ParaSwap SDK: contract calling methods: %s',
+  (testName, fetcher, contractCaller) => {
+    let paraSwap: ApproveTokenFunctions<any> & GetSpenderFunctions;
+
+    beforeAll(() => {
+      paraSwap = constructPartialSDK(
+        { network, fetcher, contractCaller },
+        constructApproveToken,
+        constructGetSpender
       );
-      const _tx = txOrError as TransactionParams;
-      const transaction = {
-        ..._tx,
-        gasPrice: '0x' + new BigNumber(_tx.gasPrice).toString(16),
-        gasLimit: '0x' + new BigNumber(5000000).toString(16),
-        value: '0x' + new BigNumber(_tx.value).toString(16),
-      };
+    });
+    test('approveToken', async () => {
+      const tx = await paraSwap.approveToken('12345', DAI);
+      await tx.wait(1);
```

Comment:
Does it work? This isn't supposed to work with `web3ContractCaller`, it doesn't return `tx with wait()`

Reply:
Somehow it does. Will have a deeper look

<img width="903" alt="Screenshot 2022-03-24 at 15 25 35" src="https://user-images.githubusercontent.com/60445720/159925897-ebee1f26-9396-4c10-9f61-b0ffc1169611.png">